### PR TITLE
add: scrolling for quiz option list

### LINKED ISSUE #408: [Feature Request] 注册答题界面增加一个下拉条
**你的功能需求和某个bug有关吗？**
`[否]`

**你想要什么样的功能？**
部分用户系统字体过大，无法选中全部选项，需要增加下拉条才可以完成答题

### PR DESCRIPTION
**Pull request by HydrogenC**
Fixes #408.  
Also fixed a long-existing bug in `PagedListView` that stuck me for over half an hour, demonstrated as followed:  

`_hasError` is set to `true` in `errorBuilder`, as we expect. 
<https://github.com/DanXi-Dev/DanXi/blob/c0037b9fb38b211e8649fc91de3d754855d9026b/lib/widget/libraries/paged_listview.dart#L223>
However, in the following lines, `_clearData` is called. 
<https://github.com/DanXi-Dev/DanXi/blob/c0037b9fb38b211e8649fc91de3d754855d9026b/lib/widget/libraries/paged_listview.dart#L228>
In which `_hasError` is restored back to `false`, overwriting the assignment on line 223. 
<https://github.com/DanXi-Dev/DanXi/blob/c0037b9fb38b211e8649fc91de3d754855d9026b/lib/widget/libraries/paged_listview.dart#L399-L404>
This causes `_hasError` to be `false` regardless of whether there is an error or not, which lead to unexpected behaviors when judging `_hasError` in `if` statements.

### CODE REVIEW

File: lib/widget/libraries/paged_listview.dart [hunk 1]
```diff
@@ -396,11 +397,13 @@ class PagedListViewState<T> extends State<PagedListView<T>>
...
   /// Clear all the data saved.
-  void _clearData() {
+  void _clearData({bool clearError = true}) {
     _data.clear();
     valueKeys.clear();
-    _hasError = false;
     _dataClearQueued = false;
+    if(clearError){
+      _hasError = true;
+    }
```
Comment (w568w) [inline lib/widget/libraries/paged_listview.dart:406]: I don't get the logic here. If `_clearData()`'s `clearError` is default to be true, shouldn't any invocation of `_clearData()` except line 229 set `_hasError = true`? Is that what you want?

Comment (HydrogenC) [reply lib/widget/libraries/paged_listview.dart:406]: I'm sorry, typo

File: lib/widget/libraries/paged_listview.dart [hunk 2]
```diff
@@ -396,11 +397,13 @@ class PagedListViewState<T> extends State<PagedListView<T>>
   }
 
   /// Clear all the data saved.
-  void _clearData() {
+  void _clearData({bool clearError = true}) {
```
Comment (w568w) [inline lib/widget/libraries/paged_listview.dart:400]: Just a suggestion: consider adding some comments here to explain why the argument `clearError` is necessary.

Comment (HydrogenC) [reply lib/widget/libraries/paged_listview.dart:402]: Resolved

**Review by w568w — APPROVED** [review]
img_v3_02ef_a6259c4c-f731-4b25-b05d-08ea5e79c37g

**Pull request merged**